tutorials-v5/time-evolution/023_v5_paper-mesolve.md

```diff
@@ -0,0 +1,575 @@
+```
+
+### Comparison of Solver Results
+
+```python
+plt.figure()
+
+plt.plot(tlist, driv_res.expect[0], "-", label="mesolve (time-dep)")
+plt.plot(tlist, driv_RWA_res.expect[0], "-.", label="mesolve (rwa)")
+# plt.plot(tlist, results_corr_fit.expect[0], '--', label=r'heomsolve')
+plt.plot(tlist, driv_br_res.expect[0], ":", linewidth=3, label="brmesolve")
+
+plt.xlabel(r"$t\, /\, \Delta^{-1}$")
+plt.ylabel(r"$\langle \sigma_z \rangle$")
+plt.legend()
+plt.show()
+```
+
+### Adiabatic Energy Switching
+
+To show where usage of these local-basis collapse operators can lead to errors, we look at a single qubit system.
+This time however, the energies of the qubit are adiabatically switched between positive and negative values.
+
+$H = \dfrac{\Delta}{2} \sin{(\omega_d t)} \sigma_z$.
+
+If the drive is slow enough, we expect the bath to respond to this change and transitions from higher to lower energy levels should be induced.
+
+```python
+# Hamiltonian
+omega_d = 0.05 * Delta  # drive frequency
+A = Delta  # drive amplitude
+H_adi = [[A / 2.0 * sigmaz(), f]]
+# H = [H0]
+
+# Bath parameters
+gamma = 0.05 * Delta / (2 * np.pi)
+
+# Simulation parameters
+tlist = np.linspace(0, 2 * T, 400)
+```
+
+```python
+# Simple mesolve
+adi_me_res = mesolve(H_adi, psi0, tlist, c_ops=c_ops_me, e_ops=e_ops)
+```
+
+```python
+# bath = UnderDampedEnvironment(lam=lambd, w0=w0, gamma=gamma_heom, T=1e-30)
+# fit_times = np.linspace(0, 5, 1000)  # range for correlation function fit
+
+# cfit, fit_info = bath.approx_by_cf_fit(fit_times,
+#                                        Ni_max=1, Nr_max=2, target_rsme=None)
+# print(fit_info["summary"])
+# Convert to a HEOM bath
+# heombath = cfit.to_bath(sigmax())
+
+
+# HEOM_corr_fit = HEOMSolver(
+# qt.QobjEvo(H),
+# heombath,
+# max_depth=max_depth,
+# options={"nsteps": 15000, "rtol": 1e-12, "atol": 1e-12},
+# )
+# results_corr_fit = HEOM_corr_fit.run(psi0 * psi0.dag(), tlist, e_ops=e_ops)
+```
+
+```python
+# BRSolve
+brme_result2 = brmesolve(H, psi0, tlist, a_ops=a_ops, e_ops=e_ops)
+```
+
+```python
+# BRSolve non-flat power spectrum
+# a_ops_non_flat = [[sigmax(), lambda w: cfit.power_spectrum(w).item()]]
+# brme_result = brmesolve(H, psi0, tlist, a_ops=a_ops_non_flat, e_ops=e_ops)
+```
+
+```python
+plt.plot(tlist, adi_me_res.expect[0], "-", label="mesolve")
+# plt.plot(tlist, results_corr_fit.expect[0], '--', label=r'heomsolve')
+# plt.plot(tlist, brme_result.expect[0], ":", linewidth=6,
+#                                           label="brmesolve non-flat")
+plt.plot(tlist, brme_result2.expect[0], ":", linewidth=6, label="brmesolve")
+
+plt.xlabel(r"$t\, /\, \Delta^{-1}$", fontsize=18)
+plt.ylabel(r"$\langle \sigma_z \rangle$", fontsize=18)
+plt.legend()
+```
+
+## References
+
+[QuTiP 5: The Quantum Toolbox in Python](about:blank)
+
+
+## About
+
+```python
+about()
+```
+
+## Testing
+
+```python
+# test sesolve gives the same result as SESolver
```

Comment:
i guess checking long-time limit of global ME is close to ground state, and that global and brmesolve produce similar results in that example, is enough?